src/clarity-angular/wizard/providers/wizard-navigation.ts

```diff
@@ -143,7 +143,7 @@ export class WizardNavigationService implements OnDestroy {
 
     public setCurrentPage(page: WizardPage): void {
         this.currentPage = page;
-// TODO: emit page onload here
+        page.onLoad.emit(page.id);
```

Comment:
`makeCurrent` is already a method on `WizardPage`. `onLoad` is an event emitter. The naming of the emitter has been carried over from 0.8 to minimize breaking changes/learning curve for users migrating from one to the other.

`setCurrentPage()` here is the final step in making a page the current page. All roads to making a page the current page (via the `Wizard`, `WizardPage`, or `WizardNavigationService`) wind up here. Per the legacy emitter, `onLoad` is emitted when a page becomes the current page.

Hopefully, that clarifies the architecture.